pkgs/by-name/dw/dwl/package.nix

```diff
@@ -0,0 +1,124 @@
+{
+  lib,
+  fetchFromGitea,
+  installShellFiles,
+  libX11,
+  libinput,
+  libxcb,
+  libxkbcommon,
+  pixman,
+  pkg-config,
+  stdenv,
+  testers,
+  wayland,
+  wayland-protocols,
+  wayland-scanner,
+  wlroots,
+  writeText,
+  xcbutilwm,
+  xwayland,
+  # Boolean flags
+  enableXWayland ? true,
+  withCustomConfigH ? (configH != null || conf != null),
+  # Configurable options
+  configH ?
+    if conf != null then lib.warn "conf parameter is deprecated; use configH instead" conf else null,
+  conf ? null,
+}:
+
+# If we set withCustomConfigH, let's not forget configH
+assert withCustomConfigH -> (configH != null || conf != null);
```

Comment:
I am thinking only `configH` should be used, since if someone decided to pass `configH = null; conf = {file}`, there would be `null` used instead of config file, but this check would pass. Maybe even asserting that if `configH` and `conf` are both set, they have to be the same, would make sense here. (or at least a warning) To prevent confusion with preference of those arguments.

Reply:
Done.